vault/vault.go

```diff
@@ -0,0 +1,114 @@
+package vault
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/hashicorp/vault/api"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+const VaultPathPrefix = "vault://"
+
+// Client to replace vault paths by the secret valued stored in the Hashicorp Vault
+type EnvVault interface {
```

Comment:
Since we don't use this interface anywhere, this is normally not implemented as an interface and Impl. Just combine this and the Impl into an `EnvVault` struct with methods. We can easily turn it back into an interface later if needed (but probably won't).